README.md

```diff
@@ -1,123 +1,323 @@
-
 ```
+4. Avro
+
+Avro files are handled in the same way as CSV files. No additional changes are required to run the pipeline.
+
+Refer [here](src/test/resources/avro) for sample data.
 
-## ReIdentification From BigQuery 
-You can. use the pipeline to read from BgQuery table and publish the re-identification data in a secure pub sub topic.
+5. CSV files with custom delimiter 
 
-Export the Standard SQL Query to read data from bigQuery
-One example from our solution guide:
+It is possible to provide csv files with custom delimiter. The delimiter has to be passed in the pipeline option as "--columnDelimiter". 
 ```
-export QUERY="select ID,Card_Number,Card_Holders_Name from \`${PROJECT_ID}.${BQ_DATASET_NAME}.100000CCRecords\` where safe_cast(Credit_Limit as int64)>100000 and safe_cast (Age as int64)>50 group by ID,Card_Number,Card_Holders_Name limit 10"
+gradle build ... -Pargs="... --columnDelimiter=|"
 ```
-Create a gcs file with the query:
+
+
+
+### S3 Scanner
+
+To use AWS S3 as a source of input files, use AWS credentials as instructed below.
+
+Export the AWS access key, secret key, and credentials provider to environment variables. 
 
 ```
-export GCS_REID_QUERY_BUCKET=<name>
-cat << EOF | gsutil cp - gs://${REID_QUERY_BUCKET}/reid_query.sql
-${QUERY}
-EOF
+export AWS_ACCESS_KEY="<access_key>"
+export AWS_SECRET_KEY="<secret_key>"
+export AWS_CRED="{\"@type\":\"AWSStaticCredentialsProvider\",\"awsAccessKeyId\":\"${AWS_ACCESS_KEY}\",\"awsSecretKey\":\"${AWS_SECRET_KEY}\"}"
 ```
-Run the pipeline by passing required parameters:
+
+Use Gradle to build and run the job to performs data loss prevention (DLP) on a CSV file stored in Amazon S3. The results will be written to BigQuery.
+
 ```
-gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=5 --maxNumWorkers=10 --runner=DataflowRunner --tableRef=<project_id>:<dataset>.<table> --dataset=<dataset> --topic=projects/<project_id>/topics/<name> --autoscalingAlgorithm=THROUGHPUT_BASED --workerMachineType=n1-highmem-4 --deidentifyTemplateName=projects/<project_id>/deidentifyTemplates/<name> --DLPMethod=REID --keyRange=1024 --queryPath=gs://<gcs_reid_query_bucket>/reid_query.sql"
+gradle spotlessApply
 
+gradle build
+
+// inspect is default as DLP Method; For deid: --DLPMethod=DEID
+gradle run -DmainClass=com.google.swarm.tokenization.DLPTextToBigQueryStreamingV2 -Pargs="--region=<region> --project=<project_id> --streaming --enableStreamingEngine --tempLocation=gs://<bucket>/temp --numWorkers=1 --maxNumWorkers=2 --runner=DataflowRunner --filePattern=s3://<bucket>>/file.csv --dataset=<name>  --inspectTemplateName=<inspect_template> --deidentifyTemplateName=<deid_tmplate> --awsRegion=<aws_region> --awsCredentialsProvider=$AWS_CRED"
 ```
+
+#### Parameters:
+
+* --awsRegion: The region where the AWS resources reside.
```

Comment:
Any limitations to the AWS region?

Reply:
Can you please check for gcp region & aws region? In some cases, the jobs need to run on the same region as the resources? If not then all good.

Reply:
Added details in https://github.com/GoogleCloudPlatform/dlp-dataflow-deidentification/blob/readme-revamp/README.md#how-to-adapt-this-pipeline-for-your-use-cases